### PR TITLE
Singleton design to controller

### PR DESCRIPTION
**Pull request by trevorNgo**
Removes null check in extension.ts and enforces singleton pattern for controller;
To follow #633

### CODE REVIEW

**Comment by SahilTara** [comment]
this is how it should have been done instead of that "Abstract" thing that we had at one point.

File: src/extension/src/controller.ts [hunk 1]
```diff
@@ -66,8 +71,25 @@ export class Controller {
...
+   * @param message The payload received from the wizard client. Message payload must include field 'module'
+   * @returns Singleton Controller type
+   */
+  public static getInstance(
+    context: vscode.ExtensionContext,
+    extensionStartTime: number
+  ) {
+    if (this._instance) {
+      this._instance.showReactPanel();
+    }
```
Comment (jcoc611-microsoft) [inline src/extension/src/controller.ts:85]: NIT: would it be more clear to have `else { this._instance = ... }` here instead of as an or statement below?

Comment (trevorNgo) [reply src/extension/src/controller.ts:85]: I agree it is es6 magic... your call

Comment (jcoc611-microsoft) [reply src/extension/src/controller.ts:85]: Yeah...you can leave it like this, it is not a big dieal

**Review by annielovesjs — DISMISSED** [review]

**Comment by SahilTara** [comment]
Hey @annielovesjs wondering if you accidentally approved this PR due to the issue you pointes out on the PR this PR depends on (#633)

**Comment by annielovesjs** [comment]
@SahilTara Hey! I was actually testing both of them individually (and tested this one twice) and this PR doesn't output the error in PR #633.

**Comment by Tanya0609** [comment]
@trevorNgo How can I check if this PR? and what are the expected result? Or I just use the wizard to generate a template to see if it breaks anything?

File: src/extension/src/controller.ts (hunk 1 above)
Comment (jsondoo) [reply src/extension/src/controller.ts:85]: I think having an else is a lot more clearer because it is a little bit confusing

Comment (jsondoo) [reply src/extension/src/controller.ts:85]: we will do it in a separate PR

**Review by jasmingoh — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Pull request merged**